lib/vmdb/settings/validator.rb

```diff
@@ -167,6 +167,42 @@ def smtp(data)
 
         return valid, errors
       end
+
+      def workers(data)
+        valid, errors = true, []
+        # worker_settings expects a hash-like structure with nested keys: :config and :workers
+        data = {:config => {:workers => data}}
+        MiqWorker.descendants.each do |worker_class|
+          result, new_errors = validate_worker_request_limit(worker_class, data)
+          if result == false
+            valid = false
+            errors += new_errors
+          end
+        end
+
+        return valid, errors
+      end
+
+      def validate_worker_request_limit(worker_class, data)
+        valid = true
+        errors = []
+        worker_settings = worker_class.worker_settings(data)
+        cpu_request     = worker_settings.fetch(:cpu_request_percent, 0)
+        cpu_limit       = worker_settings.fetch(:cpu_threshold_percent, Float::INFINITY)
+        memory_request  = worker_settings.fetch(:memory_request, 0.bytes)
+        memory_limit    = worker_settings.fetch(:memory_threshold, Float::INFINITY.megabytes)
+
+        if cpu_request > cpu_limit
+          valid = false
+          errors << [:invalid_value, "#{worker_class.settings_name}: cpu_request_percent: #{cpu_request} cannot exceed cpu_threshold_percent: #{cpu_limit}"]
```

Comment:
The key here should _technically_ be the specific key that is invalid.  We can probably pick one?

Reply:
Good 👁️ .  I did this only because it looks awful when the UI presents it (see the screenshots for the current `invalid_value`).

Reply:
Note, the code in `validate` does a similar thing as `activerecord errors.add` does for providing a section prefix + whatever message we put here.  Unfortunately, the UI tries to humanize the `cpu_threshold_percent` value into `Cpu Threshold Percent` and it looks bad so I'd rather put the actual problem in the body of the message and not in the prefix.

Reply:
Only problem there is i18n - I _think_ all of the existing keys should be in the dictionary

Reply:
Ok, I'll use the ugly format

Reply:
Ok, this is fixed, see updated screenshots and tests.